src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -168,12 +173,22 @@ private CloudWatchCollector(JSONObject config, AmazonCloudWatchClient client) th
     }
 
     /**
-     * Check if a metric should be used according to `aws_dimension_select`
+     * Check if a metric should be used according to `aws_dimension_select` or `aws_dimension_select_regex`
      */
     private boolean useMetric(MetricRule rule, Metric metric) {
-      if (rule.awsDimensionSelect == null) {
+      if (rule.awsDimensionSelect == null && rule.awsDimensionSelectRegex == null) {
+        return true;
+      }
+      if (metricsIsInAwsDimensionSelect(rule, metric) || metricIsInAwsDimensionSelectRegex(rule, metric)) {
```

Comment:
Why should we disallow that users can specify some AWS metrics with an exact string match and some metrics with an regex match? 

For example the following configuration to get all load balancers metrics with the name `myelb1`, `myelb2` or the prefix `anotherelb`:
```
{ 
  "region": "eu-west-1",
  "delay_seconds": 600,
  "metrics": [
    {
      "aws_namespace": "AWS/ELB", "aws_metric_name": "HealthyHostCount",
      "aws_dimensions": ["LoadBalancerName"], "aws_statistics": ["Minimum"],
      "aws_dimension_select": {
        "LoadBalancerName": ["myelb1", "myelb2"]
      },
      "aws_dimension_select_regex": {
        "LoadBalancerName": ["anotherelb(.*)"]
      },
      "period_seconds": 300
    }
  ]
}
```